tools/topology/topology2/sof-ace-tplg/CMakeLists.txt

```diff
@@ -17,8 +17,9 @@ DEEPBUFFER_FW_DMA_MS=100"
 
 # SDW + DMIC topology with passthrough pipelines
 # We will change NUM_HDMIS to 3 once HDMI is enabled on MTL RVP
-"cavs-sdw\;sof-mtl-rt711-4ch\;PLATFORM=mtl,NUM_DMICS=4,DMIC0_ID=2,DMIC1_ID=3,NUM_HDMIS=0,\
-PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-mtl-rt711-4ch.bin,DEEPBUFFER_FW_DMA_MS=100"
+"cavs-sdw\;sof-mtl-rt711-4ch\;PLATFORM=mtl,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,\
+DMIC0_ID=2,DMIC1_ID=3,NUM_HDMIS=0,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-mtl-rt711-4ch.bin,\
```

Comment:
The PCM IDs seem weird, and we've never exposed the second DMIC device

````
ubuntu@sh-adlp-rvp-sdw-03:~$ arecord -l
**** List of CAPTURE Hardware Devices ****
card 0: sofsoundwire [sof-soundwire], device 1: Jack in (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofsoundwire [sof-soundwire], device 10: DMIC (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
````

the list of PCM devices we use for SoundWire devices is this:

````
PCM0   playback	Jack Out
PCM1   capture 	Jack In
PCM2   playback	Speaker
PCM3   capture	Amplifier Reference (typically based on I/V feedback)
PCM4   capture 	Microphone
PCM5   playback	HDMI 1
PCM6   playback	HDMI 2
PCM7   playback	HDMI 3
PCM8   playback	HDMI 4
PCM9   RESERVED
PCM10  capture	DMIC
PCM11  capture	DMIC16kHz
PCM12  capture 	BufferedMic (aka KPB)
PCM13  playback Bluetooth
PCM14  capture 	Bluetooth

````

Reply:
@plbossart I cannot change the PCM ID or the second DMIC in this PR. This is simply to make sure that the PCM1 mics are enabled for 4ch capture

Reply:
Why can't you change those PCM IDs @ranj063 ? They are not quite right and incompatible with existing platforms.
You can do this in a second patch if you want to, but the existing result is just wrong as it stands.

Reply:
@plbossart im confused. what is the exact change you're looking for? The PCM ID's look correct to me based on your lsit no? PCM 1 for Jack 1 and PCM10 for DMIC

Reply:
this is what I object to:
line 21 'DMIC0_ID=2,DMIC1_ID=3'

Reply:
But those arent the PCM ID's. @bardliao these are the link IDs in the machine driver isnt it?

Reply:
> But those arent the PCM ID's. @bardliao these are the link IDs in the machine driver isnt it?

Yes, they are dai link IDs. The pcm IDs are DMIC0_PCM_ID, etc.

Reply:
Wow, now I am completely confused, how can the values 2 and 3 work dailink indices of the DMICs for SoundWire platforms.

What a mess.

Reply:
@plbossart sof-mtl-rt711-4ch is for MTL RVP which doesn't have amp. I.e. Only rt711 is connected. So, the DMIC dai link IDs starts from 2.

Reply:
with topology1 the SoundWire + DMIC topologies were in a different file. Looks like for topology2 we are using a single file, so no wonder things are confusing - different conventions are used.

It's a mess.

Reply:
Why are not using teh same conventions and link IDs as before ? Whats the userspace impact if we align them ?